### PR TITLE
Moved `find_manifest` functions from `sway_util` into `forc_util`

### LINKED ISSUE #955: Functions dealing with `manifest` files shouldn't be in `sway-util`
As discussed in #945 

>I think we should move these functions into the `forc-util` crate as `sway-core` is unaware of the concept of a manifest file, and only deals with source code.

- `find_parent_dir_with_file()`
- `find_manifest_dir()`
- `find_cargo_manifest_dir()`

### PR DESCRIPTION
**Pull request by eureka-cpu**
Closes #955 

I've moved the functions dealing with finding `manifest` files from `sway_util` to `forc_util` and updated the corresponding use and imports of said functions.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1568,7 +1579,7 @@ dependencies = [
  "async-std",
  "async-trait",
  "cfg-if 1.0.0",
- "dashmap",
+ "dashmap 4.0.2",
```
Comment (adlerjohn) [inline Cargo.lock:1582]: It looks like `dashmap` `v4.0.2` is used alongside `v5.2.0`. Is that an issue? cc @mitchmindtree

Comment (eureka-cpu) [reply Cargo.lock:1582]: Saw that, it got pulled in when I merged in `master`

**Review by JoshuaBatty — CHANGES_REQUESTED** [review]
lgtm, can you resolve the Cargo.lock conflict so we can merge?

File: Cargo.lock (hunk 1 above)
Comment (eureka-cpu) [reply Cargo.lock:1582]: @JoshuaBatty do you know if this would be an issue as well?

Comment (JoshuaBatty) [reply Cargo.lock:1582]: We are using an outdated version of `dashmap` in sway-lsp. https://github.com/FuelLabs/sway/blob/d725fcbaa35dbca1723d62836367d5309243a685/sway-lsp/Cargo.toml#L12

We can update that in a separate PR

File: sway-lsp/Cargo.toml [hunk 2]
```diff
@@ -10,6 +10,7 @@ description = "LSP server for Sway."
 
 [dependencies]
 dashmap = "4.0.2"
+forc-util = { version = "0.6.1", path = "../forc-util" }
```
Comment (adlerjohn) [inline sway-lsp/Cargo.toml:13]:
```suggestion
forc-util = { version = "0.7.0", path = "../forc-util" }
```

**Review by adlerjohn — CHANGES_REQUESTED** [review]

**Review by adlerjohn — APPROVED** [review]

**Review by JoshuaBatty — APPROVED** [review]

**Pull request merged**